redis/src/cluster_async/mod.rs

```diff
@@ -393,13 +392,15 @@ pub(crate) struct ClusterConnInner<C> {
     #[allow(clippy::complexity)]
     in_flight_requests: stream::FuturesUnordered<Pin<Box<Request<C>>>>,
     refresh_error: Option<RedisError>,
-    // A flag indicating the connection's closure and the requirement to shut down all related tasks.
-    shutdown_flag: Arc<AtomicBool>,
+    // Handler of the periodic topology check task.
+    topology_check_task: Option<tokio::task::JoinHandle<()>>,
```

Comment:
since it already contains more logic other than topology checks (e.g. pubusb) and we might add more,
please rename to:
topology_check_task -> periodic_checks_handler